rewrite-java-test/src/test/java/org/openrewrite/java/cleanup/ReplaceIsPresentWithIfPresentTest.java

```diff
@@ -0,0 +1,211 @@
+                            }
+                        }
+                    }
+                    """
+            )
+        );
+    }
+
+    @Test
+    void replace() {
+        rewriteRun(
+            java(
+                """
+                    package com.foobar;
+                    import java.util.Optional;
+                    import java.util.ArrayList;
+                    import java.util.List;
+                    public class A {
+                        void method() {
+                            List<Integer> list = new ArrayList<>();
+                            Optional<Integer> o = Optional.of(2);
+                            if(o.isPresent()){
+                                list.add(o.get());
+                            }
+                        }
+                    }
+                    """,
+                """
+                    package com.foobar;
+                    import java.util.Optional;
+                    import java.util.ArrayList;
+                    import java.util.List;
+                    public class A {
+                        void method() {
+                            List<Integer> list = new ArrayList<>();
+                            Optional<Integer> o = Optional.of(2);
+                            o.ifPresent((obj) -> {
+                                list.add(obj);
+                            });
+                        }
+                    }
+                    """
+            )
+        );
+    }
+
+    @Test
+    void replaceNestedIf() {
+        rewriteRun(
+            java(
+                """
+                    package com.foobar;
+                    import java.util.Optional;
+                    import java.util.ArrayList;
+                    import java.util.List;
+                    public class A {
+                        void method() {
+                            boolean x = true;
+                            List<Integer> list = new ArrayList<>();
+                            Optional<Integer> o = Optional.of(2);
+                            if(x){
+                                if(o.isPresent()){
+                                    list.add(o.get());
+                                }
+                            }
+                        }
+                    }
+                    """,
+                """
+                    package com.foobar;
+                    import java.util.Optional;
+                    import java.util.ArrayList;
+                    import java.util.List;
+                    public class A {
+                        void method() {
+                            boolean x = true;
+                            List<Integer> list = new ArrayList<>();
+                            Optional<Integer> o = Optional.of(2);
+                            if(x){
+                                o.ifPresent((obj) -> {
+                                    list.add(obj);
+                                });
```

Comment:
Could we chain [LambdaBlockToExpression](https://docs.openrewrite.org/reference/recipes/java/cleanup/lambdablocktoexpression) to get rid of the lambda block?